### ISSUE TITLE
Fix for spawning bufferedNodeProcess

### ISSUE THREAD

**Issue by benjaminRomano** [AtomLinter/atom-linter#21]
@steelbrain 
I fixed the problem in https://github.com/AtomLinter/linter-jshint/issues/151. You need to have an options object and pass it into bufferedNodeProcess : 
```
    options =
      env:
        ATOM_SHELL_INTERNAL_RUN_AS_NODE: '1'

   spawnedProcess = new BufferedNodeProcess({command, args, options, stdout, stderr, exit})
```

I can create a PR for the atom-linter code if you would like.

**Comment by steelbrain** [comment]
Doesn't BufferedNodeProcess do this already? https://github.com/atom/atom/blob/master/src/buffered-node-process.coffee#L49

**Comment by benjaminRomano** [comment]
Yeah you are right. The problem is the default options.env. Using process.env doesn't work and when I set the env myself it doesn't do this line of code. https://github.com/atom/atom/blob/master/src/buffered-node-process.coffee#L48. 

I'm currently going through each property on process.env and seeing which one is causing the problem and I'll report back when I figure that out

**Comment by steelbrain** [comment]
Yep, it makes it work, but then the `.jshintrc` finding doesn't work, unfortunately.

**Comment by benjaminRomano** [comment]
Why does it break the ```.jshintrc```?

**Comment by steelbrain** [comment]
'cause `.jshintrc` is searched in `HOMEDRIVE`, `HOMEPATH`, `HOMEDIR` and `HOME` env vars. and we remove them when we provide an object which has only the `ATOM_SHELL_INTERNAL_RUN_AS_NODE` value

**Comment by benjaminRomano** [comment]
I just tested adding HOMEDRIVE and HOMEDIR to the options.env from process.env and the code still works on windows. 

For a temporary fix, we could have windows specific code that creates an options object with just those properties. However, it would be ideal if we could figure out which exact property is causing things to fail.

**Comment by steelbrain** [comment]
I am like super-stretched up between work and I would highly appreciate it if someone narrows the error down to a single env var.

**Comment by steelbrain** [comment]
Fixed by #22
